.github/workflows/continuous-build.yml

```diff
@@ -20,49 +21,79 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
           go-version: '^1.15'
+
       - uses: actions/cache@v2
         with:
           path: ~/go/pkg/mod
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
+
       - name: Run tests
         run: go test -cover ./...
+
       - name: Build daemon
         run: go build ./cmd/tracing
-  build_docker:
-    name: Build Docker image
+  build_publish_daemon_image:
+    name: Build and Publish X-Ray daemon docker image to docker hub and ECR
     needs: build
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
-      - name: Login to DockerHub
+
+      - name: Install latest version of aws cli
+        run: |
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          sudo ./aws/install
+          aws --version
+
+      - name: Configure AWS Credentials
         if: ${{ github.event_name == 'push' }}
-        uses: docker/login-action@v1
+        uses: aws-actions/configure-aws-credentials@v1
         with:
-          username: ${{ secrets.DOCKER_USER }}
-          password: ${{ secrets.DOCKER_PASS }}
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        if: ${{ github.event_name == 'push' }}
+        run: aws ecr-public get-login-password --region us-east-1 | docker login -u AWS --password-stdin public.ecr.aws
+
+#      - name: Login to DockerHub
```

Comment:
We need to continue to push to docker hub right? Why is this commented out?

Reply:
yes I am testing docker login action but having issues with it. I will add it once I am done with tests.

Reply:
How are you testing? PR won't have secrets